### DISCUSSION TITLE
Is there a simple way I can extract agent output from Autogen? Or save the output to local drive?

### DISCUSSION THREAD

**Discussion by mraguth** [microsoft/autogen#1846] [Q&A] [answered]
I'm not requesting any code; my intention is to obtain the output generated by Autogen. If the output is in text format, I am comfortable saving it in a web-friendly format. However, if the output is in 'dataframe' format, I would prefer to save it as either parquet or .csv files. I am willing to create a function if necessary.

I have a great appreciation for Autogen and its capabilities. However, when it comes to real work situations, it becomes essential to merge local datasets in order to derive meaningful insights. Without the ability to converge internal data or extract the data output from agents, progress of the Autogen can be hindered significantly.

**Comment by ekzhu** [comment, marked as answer]
`chat_result = proxy.initiate_chat(assistant, message="...")`

You can take a look at the `chat_result` object which contains the output, including messages, summary, and token usage. 

If you are looking for logging, you can take a look at the logging example: https://github.com/microsoft/autogen/blob/main/notebook/agentchat_logging.ipynb

**Reply by mraguth** [reply]
Thank you. The response column from the dataset will be good enough for me to extract Agent output data.

**Comment by wmwxwa** [comment]
@mraguth Would you mind elaborate on the use case and results you are trying to achieve? If you do you feel comfortable discussing it in public, please reach out to wangwilliam@microsoft.com

**Reply by mraguth** [reply]
Our designed agents are providing valuable insights to us. However, to extract all those insights is getting hard for us. In case you need more insights, I am happy to write to your email. Thank you.

**Reply by sonichi** [reply]
@mraguth to be clear, @wmwxwa is not a contributor of the AutoGen project.
If you intend to contact the maintainers, please join our discord server and DM me.

**Reply by wmwxwa** [reply]
@mraguth We are working on a connector for AutoGen and Azure Cosmos DB. The latter stores raw and vectorized data from agent interactions together as JSON documents. OpenAI currently uses Cosmos DB to run ChatGPT's data tier. Would that help?

**Reply by wmwxwa** [reply]
@mraguth The AutoGen Discord server that Chi mentioned is here: https://discord.gg/9yzr6wUx

**Reply by mraguth** [reply]
Thank you.

**Reply by mraguth** [reply]
Hi @sonichi  Currently, our agent queries the structured data and writes the top 10 bullet points as data insights. Now we have a option to cut and paste those from the output window and paste it to pdf and send it out. However, we are looking for a more efficient way to collect and communicate these points to our customers.

One potential solution is to leverage a database or a similar system to store these findings. Rather than relying on manual copying and pasting, we can directly point these insights to the database. This approach would not only streamline the process but also ensure scalability as we continue to generate more insights. BTW, I like your presentation.
